tools/proofers/dp_proof.js

```diff
@@ -230,3 +230,17 @@ function transformText(transformType) {
     txtArea.focus();
 }
 
+// Avoid submitting twice if user double-clicks button
+// Usage: onClick="return submitJustOnce(this);"
+//
+// Uses a data attribute to store when button has been clicked for the first time
+// Returns true on first click to continue submission
+// If attribute is already set (second click), return false to avoid re-submitting
+function submitJustOnce(btn) {
+    if (btn.dataset.alreadysubmitted) {
+        return false;
+    } else {
+        btn.dataset.alreadysubmitted = true;
+        return true;
```

Comment:
Is `btn.dataset.alreadysubmitted = true;` necessary, or could this part of the conditional just be `return true;`?

Reply:
That's the flag we need to track if the button has been clicked already.